packages/tofu/src/inspectSource.js

```diff
@@ -183,37 +183,110 @@ function inspectGlobals(
  * @param {import('@babel/types').Node} ast
  * @param {string[]} [packagesToInspect] - List of module IDs to look for; if
  *   omitted, will inspect all imports
+ * @param {boolean} [deep] Return named exports and imports
  * @returns {string[]} - List of imported identifiers
  */
-function inspectEsmImports(ast, packagesToInspect) {
+function inspectEsmImports(ast, packagesToInspect, deep = true) {
   const shouldInspectAll = !packagesToInspect
   const pkgsToInspect = new Set(packagesToInspect)
 
   /** @type {Set<string>} */
-  const esmImports = new Set()
+  const result = new Set()
 
-  /**
-   * @param {import('@babel/traverse').NodePath<
-   *   | import('@babel/types').ImportDeclaration
-   *   | import('@babel/types').ExportNamedDeclaration
-   * >} path
-   */
-  const handleNodePath = (path) => {
-    const importSource = path.node.source?.value
+  traverse(ast, {
+    ImportDeclaration: (path) => {
+      const importSource = path.node.source.value
+      if (!shouldInspectAll && !pkgsToInspect.has(importSource)) {
+        return
+      }
 
-    if (importSource && (shouldInspectAll || pkgsToInspect.has(importSource))) {
-      esmImports.add(importSource)
-    }
-  }
+      if (!deep) {
+        result.add(importSource)
+        return
+      }
+
+      const { specifiers } = path.node
+
+      const hasNamespaceImport = specifiers.some(
+        (spec) => spec.type === 'ImportNamespaceSpecifier'
+      )
+
+      const hasOnlyDefaultImport = specifiers.every(
+        (spec) => spec.type === 'ImportDefaultSpecifier'
+      )
+
+      if (
+        hasNamespaceImport ||
+        hasOnlyDefaultImport ||
+        specifiers.length === 0
+      ) {
+        // import * as foo from 'bar', import foo from 'baz', import 'quux'
+        result.add(importSource)
+      } else {
+        // Mixed or named imports
+        specifiers.forEach((spec) => {
+          if (spec.type === 'ImportSpecifier') {
+            // it can be an identifier or a string literal, to my amazement
+            const importedName =
+              spec.imported.type === 'Identifier'
+                ? spec.imported.name
+                : spec.imported.value
+            result.add(`${importSource}.${importedName}`)
+          } else if (spec.type === 'ImportDefaultSpecifier') {
+            // import foo, {bar} from 'baz'
+            result.add(importSource)
+          }
+        })
+      }
+    },
 
-  traverse(ast, {
     ExportNamedDeclaration: (path) => {
```

Comment:
`ExportAllDeclaration` also exists

Reply:
I'm not sure what this is.  This handler works for `export * as name from 'pkg'`, as that's an `ExportNamedDeclaration`.